spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/pubsub/GcpPubSubAutoConfiguration.java

```diff
@@ -130,8 +130,11 @@ public PubSubPublisherTemplate pubSubPublisherTemplate(PublisherFactory publishe
 
 	@Bean
 	@ConditionalOnMissingBean
-	public PubSubSubscriberTemplate pubSubSubscriberTemplate(SubscriberFactory subscriberFactory) {
-		return new PubSubSubscriberTemplate(subscriberFactory);
+	public PubSubSubscriberTemplate pubSubSubscriberTemplate(SubscriberFactory subscriberFactory,
+			ObjectProvider<PubSubMessageConverter> pubSubMessageConverter) {
+		PubSubSubscriberTemplate pubSubSubscriberTemplate = new PubSubSubscriberTemplate(subscriberFactory);
+		pubSubMessageConverter.ifUnique(pubSubSubscriberTemplate::setMessageConverter);
```

Comment:
Any possible unit-test to cover this change?

And your name to the `@author` list 😄 

Thanks

Reply:
Added author tag.

I tried writing a unit test but it seemed a little bit messy since I would have to create a bunch of mocks to do it properly. I thought this was a signal that the configuration would be better tested via Integration test. What are your thoughts on this?

Thanks

Reply:
I think you just need to add extra `@Configuration` into the `PubSubTemplateIntegrationTests` with the `PubSubMessageConverter` `@Bean` and verify in the existing (or new) test-case against that `this.contextRunner` to be sure that `PubSubTemplate` is supplied with your  `PubSubMessageConverter`.